### PR TITLE
feat(geo): Forward ref to react-map-gl component

### PR DESCRIPTION
**Pull request by slaymance**
#### Description of changes
`react-map-gl` supports passing a `ref` to the `Map` component to grant access to the underlying map object to add event callbacks and such ([see the docs](https://visgl.github.io/react-map-gl/docs/api-reference/map#methods)). This PR provides this functionality for users of `MapView` by forwarding the `ref` prop to the wrapped component and updating the TypeScript type.

#### Issue #, if available
#1790 

#### Description of how you validated changes
Ensuring e2e tests function is sufficient validation.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] Relevant documentation is changed or added (and PR referenced) - N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .changeset/nine-pillows-look.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@aws-amplify/ui-react': minor
+---
+
+Add forward ref functionality to MapView
```
Comment (slaymance) [inline .changeset/nine-pillows-look.md:5]: Minor makes sense here? The `ref` property was not valid prior to this change so that was my reasoning.

Comment (zchenwei) [reply .changeset/nine-pillows-look.md:5]: Right, we publish `ref` on other primitives as minor

**Review by zchenwei — COMMENTED** [review]
Overall, I am good with it. Can we add a unit test to make sure the `ref` prop gets passed through? Like how we do in primitive tests

File: packages/react/src/components/Geo/MapView/index.tsx [hunk 2]
```diff
@@ -21,66 +21,64 @@ export type MapViewProps = Omit<MapProps, 'transformRequest'>;
...
-      amplifyConfig.geo?.AmazonLocationService ??
-      {},
-    [amplifyConfig]
-  );
-  const [transformRequest, setTransformRequest] = useState<
-    TransformRequestFunction | undefined
-  >();
+export const MapView = forwardRef<MapRef, MapViewProps>(
+  ({ mapLib, mapStyle, style, ...props }, ref) => {
+    const amplifyConfig = Amplify.configure() as any;
```
Comment (reesscot) [inline packages/react/src/components/Geo/MapView/index.tsx:26]: non-blocking: can we do anything to improve the typing here?

**Review by reesscot — APPROVED** [review]
LGTM

**Comment by slaymance** [comment]
I don't think unit testing this will be straightforward since the map needs a functioning Amplify config to render.

**Review by calebpollman — APPROVED** [review]
LGTM 🚢

**Review by zchenwei — APPROVED** [review]

**Pull request merged**